### PR TITLE
fix(clerk-react): Fix forceRedirectUrl and fallbackRedirectUrl when passed to button components

### PR DESCRIPTION
**Pull request by BRKalow**
## Description

When passing `forceRedirectUrl` and `fallbackRedirectUrl` to `SignInButton` and `SignUpButton`, the props were getting renamed and subsequently ignored by the flow when opening in a modal. I fixed the property names and added a few test cases to cover the behavior.

Note that the non-modal flow wasn't broken as it expects the prefixed props (`signInForceRedirectUrl`, `signInFallbackRedierctUrl`).

Writing the tests also uncovered a bug with our modal implementation where the entire modal DOM tree was wrapped in an element that had `aria-hidden`, which I fixed.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/react/src/components/SignInButton.tsx [hunk 1]
```diff
@@ -11,21 +12,27 @@ export const SignInButton = withClerk(({ clerk, children, ...props }: WithClerkP
   const child = assertSingleChild(children)('SignInButton');
 
   const clickHandler = () => {
-    const opts = {
+    const opts: SignInProps = {
+      forceRedirectUrl,
+      fallbackRedirectUrl,
```
Comment (BRKalow) [inline packages/react/src/components/SignInButton.tsx:17]: The modal mode relies on the non-prefixed props, adding the type here revealed the issue.

File: packages/react/src/components/SignInButton.tsx [hunk 2]
```diff
@@ -11,21 +12,27 @@ export const SignInButton = withClerk(({ clerk, children, ...props }: WithClerkP
...
     };
 
     if (mode === 'modal') {
       return clerk.openSignIn(opts);
     }
-    return clerk.redirectToSignIn(opts);
+    return clerk.redirectToSignIn({
+      ...opts,
+      signInFallbackRedirectUrl: fallbackRedirectUrl,
+      signInForceRedirectUrl: forceRedirectUrl,
```
Comment (BRKalow) [inline packages/react/src/components/SignInButton.tsx:28]: when calling the redirect methods, the prefixed props are necessary. Added here.

File: packages/react/src/components/SignInButton.tsx [hunk 3]
```diff
@@ -11,21 +12,27 @@ export const SignInButton = withClerk(({ clerk, children, ...props }: WithClerkP
...
+    });
   };
 
   const wrappedChildClickHandler: React.MouseEventHandler = async e => {
-    await safeExecute((child as any).props.onClick)(e);
+    if (child && typeof child === 'object' && 'props' in child) {
+      await safeExecute(child.props.onClick)(e);
+    }
```
Comment (BRKalow) [inline packages/react/src/components/SignInButton.tsx:35]: Some auto-fix of a lint rule caused a type error, so I tweaked the logic a bit here as a result.

File: packages/clerk-js/src/ui/elements/Modal.tsx [hunk 4]
```diff
@@ -50,15 +50,13 @@ export const Modal = withFloatingTree((props: ModalProps) => {
     >
       <ModalContext.Provider value={modalCtx}>
         <Flex
-          aria-hidden
```
Comment (BRKalow) [inline packages/clerk-js/src/ui/elements/Modal.tsx:53]: I don't think we want this, as it removes the entire modal from the accessibility tree.

**Review by panteliselef — APPROVED** [review]

**Review by tmilewski — APPROVED** [review]

File: packages/react/src/components/__tests__/SignUpButton.test.tsx [hunk 5]
```diff
@@ -63,6 +63,7 @@ describe('<SignUpButton/>', () => {
     userEvent.click(btn);
     await waitFor(() => {
       expect(mockRedirectToSignUp).toHaveBeenCalledWith({
+        fallbackRedirectUrl: url,
         signUpFallbackRedirectUrl: url,
```
Comment (panteliselef) [inline packages/react/src/components/__tests__/SignUpButton.test.tsx:67]: @BRKalow  why do we need both ?

File: packages/clerk-js/src/ui/elements/Modal.tsx (hunk 4 above)
Comment (JacobMGEvans) [reply packages/clerk-js/src/ui/elements/Modal.tsx:53]: Definitely not 😅

File: packages/react/src/components/SignInButton.tsx (hunk 3 above)
Comment (JacobMGEvans) [reply packages/react/src/components/SignInButton.tsx:35]: Interesting.

File: packages/react/src/components/__tests__/SignUpButton.test.tsx (hunk 5 above)
Comment (panteliselef) [reply packages/react/src/components/__tests__/SignUpButton.test.tsx:67]: I believe types only require the prefixed variant, so we are good

**Review by dimkl — APPROVED** [review]

**Pull request merged**